File: algorithms/coding-interview-questions-blind75-lol/1/two_sum.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> m;
        int complement;
        for(int i = 0; i < size(nums); ++i) {
            complement = target - nums[i];
            if (m.count(complement))
                return {i, m[complement]};
            m[nums[i]] = i;
        }
        return {};
    }
};

File: algorithms/blind75/1/two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        m = {}
        for i, x in enumerate(nums):
            complement = target - x
            if complement in m:
                return [i, m[complement]]
            m[x] = i
